yt/utilities/lib/grid_traversal.pxd

```diff
@@ -29,11 +29,26 @@ ctypedef void sampler_function(
                 int index[3],
                 void *data) nogil
 
+#-----------------------------------------------------------------------------
+# walk_volume(VolumeContainer *vc,  np.float64_t v_pos[3], np.float64_t v_dir[3], sampler_function *sample,
+#             void *data, np.float64_t *return_t = NULL, np.float64_t max_t = 1.0)
+#      vc        VolumeContainer*  : Pointer to the volume container to be traversed.
+#      v_pos     np.float64_t[3]   : The x,y,z coordinates of the ray's origin.
+#      v_dir     np.float64_t[3]   : The x,y,z coordinates of the ray's direction.
+#      sample    sampler_function* : Pointer to the sample function to be used.
+#      return_t  np.float64_t*     : # TODO: Unsure of behavior. Defaulted to NULL.
```

Comment:
What is the ```return_t```?
I see this:
```if return_t != NULL: return_t[0] = exit_t```
So is this a variant of ```ObjectList[X][Y][Z]``` I see in A & W's paper?

Reply:
This is to let us know what the final `t` value is that's still inside the volume container.  So if it walks off the end at t=0.4 or something, it passes that info back.